error: Conversion to 'std::size_t' from 'const int' when building ad_map_access

- Platform: Ubuntu18.04
- command line:

```
➜  map git:(master) colcon build     
Starting >>> spdlog  
Starting >>> ad_map_opendrive_reader
Finished <<< ad_map_opendrive_reader [0.29s]                    
Finished <<< spdlog [0.32s]
Starting >>> ad_physics
Finished <<< ad_physics [0.28s]                     
Starting >>> ad_map_access
--- stderr: ad_map_access                             
/path/to/map/ad_map_access/impl/tests/route/RoutePlanningTests.cpp: In member function ‘void RoutePlanningTest::validateRouteConnections(const ad::map::route::FullRoute&, RoutePlanningTest::RouteBehaviour)’:
/path/to/map/ad_map_access/impl/tests/route/RoutePlanningTests.cpp:218:16: error: conversion to ‘std::size_t {aka long unsigned int}’ from ‘const RouteLaneOffset {aka const int}’ may change the sign of the result [-Werror=sign-conversion]
       if (j != static_cast<size_t>(route.roadSegments[i].drivableLaneSegments[j].routeLaneOffset))
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [generated/ad_map_access_unit_tests/CMakeFiles/ad_map_access_tests.dir/route/RoutePlanningTests.cpp.o] Error 1
make[1]: *** [generated/ad_map_access_unit_tests/CMakeFiles/ad_map_access_tests.dir/all] Error 2
make: *** [all] Error 2
---
Failed   <<< ad_map_access [4.80s, exited with code 2]
                                
Summary: 3 packages finished [5.59s]
  1 package failed: ad_map_access
  1 package had stderr output: ad_map_access
  4 packages not processed
```

Thanks in advance!

When building the MapMaker, I came across another error as follows:

```
➜  map git:(master) ✗ colcon build --cmake-args -DBUILD_TESTING=ON 
Starting >>> spdlog  
Starting >>> ad_map_opendrive_reader
Finished <<< spdlog [0.39s]                                
Starting >>> ad_physics
Finished <<< ad_map_opendrive_reader [0.39s]
Finished <<< ad_physics [0.38s]                     
Starting >>> ad_map_access
Finished <<< ad_map_access [0.64s]                     
Starting >>> ad_map_access_qgis_python
Starting >>> MapMaker
Starting >>> ad_map_access_test_support
--- stderr: MapMaker                 
CMake Warning (dev) at /usr/src/googletest/googlemock/CMakeLists.txt:10 (option):
  Policy CMP0077 is not set: option() honors normal variables.  Run "cmake
  --help-policy CMP0077" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  For compatibility with older versions of CMake, option is clearing the
  normal variable 'BUILD_SHARED_LIBS'.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at geometry/CMakeLists.txt:15 (message):
  Unable to find lapacke.  Maybe you need to install liblapacke-dev and
  libopenblas-dev.  Set LAPACK_LIB.


---
Failed   <<< MapMaker [4.83s, exited with code 1]
Aborted  <<< ad_map_access_qgis_python [10.0s]   
Aborted  <<< ad_map_access_test_support [10.4s]
                                 
Summary: 4 packages finished [12.0s]
  1 package failed: MapMaker
  2 packages aborted: ad_map_access_qgis_python ad_map_access_test_support
  2 packages had stderr output: MapMaker ad_map_access_qgis_python
  1 package not processed
```

So I tried to `sudo apt install liblapacke-dev`, and everything is okay finally :smiley:

--- a/ad_map_access/impl/tests/route/RoutePlanningTests.cpp
+++ b/ad_map_access/impl/tests/route/RoutePlanningTests.cpp
@@ -215,7 +215,7 @@ void RoutePlanningTest::validateRouteConnections(route::FullRoute const &route, 
                   - route.roadSegments[i].drivableLaneSegments[0].routeLaneOffset)
         << "i:" << i << " j:" << j << " route:" << route;
 
-      if (j != static_cast<size_t>(route.roadSegments[i].drivableLaneSegments[j].routeLaneOffset))
+      if (static_cast<int32_t>(j) != route.roadSegments[i].drivableLaneSegments[j].routeLaneOffset)
       {
         EXPECT_NE(RouteIsConstant, routeBehavior);
       }

--- a/doc/BUILDING.md
+++ b/doc/BUILDING.md
@@ -28,6 +28,7 @@ The components within this repository have some dependencies:
    - Python lib
  - **map_maker** (in tools):
    - *ad_map_access*
+   - LAPACK
    - Osmium >= 2.13: <https://osmcode.org/libosmium/>
  - ***all components when enabling unit tests***:
    - gtest aka. googletests < 1.10 : <https://github.com/google/googletest>
@@ -40,10 +41,11 @@ Dependencies provided by Ubunutu (>= 16.04):
  - gtest
  - libpython-dev
  - libosmium2-dev
+ - liblapacke-dev
 
 Those can be installed by calling:
 ```bash
-$>  sudo apt-get install libboost-all-dev libpugixml-dev libproj-dev libgtest-dev libpython-dev libosmium2-dev
+$>  sudo apt-get install libboost-all-dev libpugixml-dev libproj-dev libgtest-dev libpython-dev libosmium2-dev liblapacke-dev
 ```
 
 Additional dependencies for the python bindings:

--- a/doc/CHANGELOG.md
+++ b/doc/CHANGELOG.md
@@ -1,4 +1,5 @@
 ## Latest changes
+* Fix build of unit tests under Ubuntu18.04 with gcc 7.5.0
 
 ## Release 2.1.0
 #### :ghost: Maintenance

